google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/ExceptionMockServerTest.java

```diff
@@ -0,0 +1,163 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spanner.connection;
+
+import static com.google.cloud.spanner.SpannerApiFutures.get;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+import com.google.cloud.spanner.MockSpannerServiceImpl.SimulatedExecutionTime;
+import com.google.cloud.spanner.MockSpannerServiceImpl.StatementResult;
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.SpannerException;
+import com.google.cloud.spanner.Statement;
+import io.grpc.Status;
+import java.util.Arrays;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class ExceptionMockServerTest extends AbstractMockServerTest {
+
+  @After
+  public void clearRequests() {
+    mockSpanner.clearRequests();
+  }
+
+  @Test
+  public void testUpdateAsyncException() {
+    Statement update = Statement.of("update foo set bar=1 where baz=1");
+    mockSpanner.putStatementResult(
+        StatementResult.exception(
+            update,
+            Status.INVALID_ARGUMENT.withDescription("Table 'foo' not found").asRuntimeException()));
+
+    try (Connection connection = createConnection()) {
+      SpannerException exception =
+          assertThrows(SpannerException.class, () -> get(connection.executeUpdateAsync(update)));
+      assertNotNull(exception.getSuppressed());
+      assertEquals(1, exception.getSuppressed().length);
+      Throwable suppressed = exception.getSuppressed()[0];
+      String methodName = new Object() {}.getClass().getEnclosingMethod().getName();
+      assertTrue(
+          Arrays.stream(suppressed.getStackTrace())
+              .anyMatch(
+                  element ->
+                      element.getClassName().equals(ExceptionMockServerTest.class.getName())
+                          && element.getMethodName().equals(methodName)));
+    }
+  }
+
+  @Test
+  public void testUpdateException() {
+    Statement update = Statement.of("update foo set bar=1 where baz=1");
+    mockSpanner.putStatementResult(
+        StatementResult.exception(
+            update,
+            Status.INVALID_ARGUMENT.withDescription("Table 'foo' not found").asRuntimeException()));
+
+    try (Connection connection = createConnection()) {
+      SpannerException exception =
+          assertThrows(SpannerException.class, () -> connection.executeUpdate(update));
+      assertNotNull(exception.getSuppressed());
+      assertEquals(0, exception.getSuppressed().length);
+    }
+  }
+
+  @Test
+  public void testQueryAsyncException() {
+    Statement update = Statement.of("select * from foo");
```

Comment:
nit: update -> select. These can be modified in other parts within this file too, and can be private static final variables within this class.

Reply:
Done